main.c

```diff
@@ -0,0 +1,212 @@
+    struct node* newnode, *temp = NULL;
+    int choice = 1;
+
+    while (choice) {
+        newnode = (struct node*)malloc(sizeof(struct node));
+        if (!newnode) {
+            printf("Memory allocation failed\n");
+            return head;
+        }
+        printf("Enter data: ");
+        scanf("%d", &newnode->data);
+        newnode->next = NULL;
+
+        if (head == NULL) {
+            head = temp = newnode;
+        } else {
+            temp->next = newnode;
+            temp = newnode;
+        }
+        (*count)++;
+        printf("Press 1 to continue and 0 to exit: ");
+        scanf("%d", &choice);
+    }
+    return head;
+}
+
+void display(struct node* head) {
+    struct node *temp = head;
+    if (head == NULL) {
+        printf("List is empty.\n");
+        return;
+    }
+    printf("Linked List elements:\n");
+    while (temp != NULL) {
+        printf("%d\n", temp->data);
+        temp = temp->next;
+    }
+    printf("NULL\n");
+}
+
+struct node* insertatbeginning(struct node* head, int* count) {
+    struct node *newnode = (struct node*)malloc(sizeof(struct node));
+    if (!newnode) {
+        printf("Memory allocation failed\n");
+        return head;
+    }
+    printf("Enter the data to insert at the beginning: ");
+    scanf("%d", &newnode->data);
+    newnode->next = head;
+    head = newnode;
+    (*count)++;
+    return head;
+}
+
+struct node* insertatend(struct node* head, int* count) {
+    struct node *newnode = (struct node*)malloc(sizeof(struct node));
+    struct node *temp = head;
+
+    if (!newnode) {
+        printf("Memory allocation failed\n");
+        return head;
+    }
+    printf("Enter the data to insert at the end: ");
+    scanf("%d", &newnode->data);
+    newnode->next = NULL;
+
+    if (head == NULL) {
+        head = newnode;
+    } else {
+        while (temp->next != NULL) {
+            temp = temp->next;
+        }
+        temp->next = newnode;
+    }
+    (*count)++;
+    return head;
+}
+
+struct node* insertatposition(struct node* head, int* count) {
+    struct node *newnode, *temp = head;
+    int position, i;
+
+    printf("Enter position to insert: ");
+    scanf("%d", &position);
+
+    if (position < 0 || position > (*count)) {
+        printf("Invalid position\n");
+        return head;
+    }
+
+    newnode = (struct node*)malloc(sizeof(struct node));
+    if (!newnode) {
+        printf("Memory allocation failed\n");
+        return head;
+    }
+    printf("Enter data: ");
+    scanf("%d", &newnode->data);
+    newnode->next = NULL;
+
+    if (position == 0) {
+        newnode->next = head;
+        head = newnode;
+    } else {
+        for (i = 0; i < position - 1 && temp != NULL; i++) {
+            temp = temp->next;
+        }
+        if (temp != NULL) {
+            newnode->next = temp->next;
+            temp->next = newnode;
+        } else {
+            printf("Position not found\n");
+            free(newnode);
+        }
+    }
+    (*count)++;
```

Comment:
Why *count ?

Reply:
*count is a memory address of the variable count, by using this, function access to the actual memory location of count and directly modify the count value.